## [Слияние с проверкой](../../../solutions/5.3/53_e.py)

Когда-то вы уже писали функцию `merge`, которая производит слияние двух отсортированных кортежей.

Давай-те её немного переработаем.

Введём систему проверок:

- если один из параметров не является итерируемым объектом, то вызовите исключение `StopIteration`;
- если значения входных параметров содержат «неоднородные» данные, то вызовите исключение `TypeError`;
- если один из параметров не отсортирован, то вызовите исключение `ValueError`.
Проверки следует проводить в указанном порядке.

Если параметры прошли все проверки, верните итерируемый объект, являющийся слиянием двух переданных.

### Примечание

В решении не должно быть вызовов требуемых функций.

### Пример 1

__Ввод__
```python
print(*merge(35, (1, 2, 3)))
```

__Вывод__
```plaintext
Вызвано исключение StopIteration
```

### Пример 2

__Ввод__
```python
print(*merge([3, 2, 1], range(10)))
```

__Вывод__
```plaintext
Вызвано исключение ValueError
```